modules/ROOT/pages/client-credentials.adoc

```diff
@@ -0,0 +1,297 @@
+|The OAuth scopes to be requested during the dance. If not provided, it defaults to those in the annotation.
+
+|`objectStore`
+|NO
+|NOT_SUPPORTED
+|N/A
+|A reference to the ObjectStore used for storing each resource owner ID’s data. If not specified, the runtime automatically provisions the default one.
+|===
+
+=== About the Use of Expressions
+
+The previous table identifies many of the synthetic parameters that accept expressions. Using expressions there has the same effect as using expressions in a regular parameter: it turns the configuration into a dynamic one.
+
+=== OAuth Connection DSL
+
+The following example shows  generated DSL:
+
+[source, java, linenums]
+----
+<sfdc:config name="salesforce">
+    <sfdc:client-credentials-connection display="PAGE" immediate="FALSE" prompt="CONSENT">
+        <sfdc:oauth-client-credentials clientId="${sfdc.consumerkey}" clientSecret="${sfdc.consumersecret}" tokenUrl="http://..." />
+        <sfdc:oauth-store-config objectStore="oauthObjectStore" />
+</sfdc:config>
+----
+
+Regular and OAuth parameters are all shown at the connection provider level, just like in any other provider.
+The parameters related to the Client Credentials grant type are placed on a child element called <oauth-client-credentials>.
+The parameters related to ObjectStore will be placed in a child element called <oauth-store-config>.
+
+=== Configuring a Custom ObjectStore
+
+The obtained access tokens are stored in an ObjectStore. By default, the SDK stores them in the app’s default store, but users can define their own custom one, as shown in this example:
+
+[source, java, linenums]
+----
+<os:object-store name="tokenStore"
+   entryTtl="1"
+   entryTtlUnit="HOURS"
+   maxEntries="100"
+   persistent="true"
+   expirationInterval="30"
+   expirationIntervalUnit="MINUTES" />
+
+<sfdc:config name="salesforce">
+    <sfdc:client-credentials-connection display="PAGE" immediate="FALSE" prompt="CONSENT">
+        <sfdc:oauth-client-credentials clientId="${sfdc.consumerkey}" clientSecret="${sfdc.consumersecret}" tokenUrl="http://..." />
+        <sfdc:oauth-store-config objectStore="tokenStore" />
+</sfdc:config>
+----
+
+== Handling Expired Client Credentials Access Token
+
+Most service providers return (or should return) access tokens with a limited lifespan. In general, access tokens usually expire about 30 to 60 minutes after being issued, although this is not a standard that is strictly enforced by providers.
+In the same way, providers often provide a refresh token with the access token. The refresh token is a second token that can be used to replace an expired access token with a fresh one, without the need to perform the OAuth dance again.
+There is not an enforced standard that the SDK can use to automatically detect a token expiration scenario and obtain a new one. Different APIs communicate that in different ways. That is why the AccessTokenExpiredException exception exists, shown below:
```

Comment:
```suggestion
There is not an enforced standard for an SDK to automatically detect an expired token and replace it. Because different APIs communicate that in different ways, SDKs use `AccessTokenExpiredException`: 
```